docs/content/docs/data-structures/counter.md

```diff
@@ -1,5 +1,208 @@
+Why incur the overhead of another runtime type?
+
+The key to the answer here is that DDSes with simpler merge strategies (like `SharedMap`) take a somewhat brute-force approach to merging concurrent edits.
+For a semantic data type like a counter, this can result in undesirable behavior.
+
+#### SharedMap Example
+
+Let's illustrate the issue with an example.
+
+Here, we will use a [SharedMap][] to store our shared integer value.
+For simplicity, it will be stored under a static key: `counter-key`.
+
+Let's say that two users are collaborating on an app with a counter widget.
+We will refer to these users as User A and User B.
+The current value of that widget is 42.
+
+What if User A and User B both simultaneously press the `+` button in their UI to increment the current value by 1?
+Behind the scenes, this increment is implemented by writing the updated value to the map.
+Each edit is then sequenced and broadcast to other collaborators.
+
+For the purpose of this example, we are ignoring the fact that two operations will never really occur at exactly the same time.
+This is intentional.
+Since increments and decrements are [commutative](https://en.wikipedia.org/wiki/Commutative_property), it shouldn't matter in what order two nearly concurrent increment operations occur; the result _should_ be the same.
+
+Here, we would expect that, after both users have processed all incoming edits, the resulting value of the counter would be 44.
+Each user pressed the `+` button once.
+42 + 1 + 1 = **44**, right?
+Again, note that we are ignoring which `+ 1` came from which user, as the order should not matter.
+
+Unfortunately, because `SharedMap` employs a _last-write-wins_ merge strategy, if User A and User B make their edits at the same time, one of the two updates will be sequnced after the other, and that value will win.
+Regardless of which user's edit was sequenced first and which was sequenced last, each user's client will update their map entry to be 42 (the current value they see) + 1 = **43**, and the latter of the two operations to be sequenced will clobber the first.
+
+So in this case, both users would see the value 43, rather than 44.
+
+This is a problem!
+If this shared counter value was being used to track store inventory, for example, that inventory would now be incorrect!
+
+The solution to this problem is a specialized DDS that tracks changes to the shared value as _increments_ and _decrements_, which can be summed together in any order and still reach eventual consistency.
+
+{{% callout tip %}}
+
+You may be asking yourself: but wouldn't the counter incrementing by 2 after the user pressed the increment button just once confuse the user?
+How do I make clear to the user that the additional increment was the result of a collaborator's actions?
+
+Check out [User presence and audience]({{< relref "audience.md" >}}) for an overview of Fluid's APIs for working with user presence!
+
+{{% /callout %}}
+
+## Creation
```

Comment:
This creation section is basically the same for all DDS. Are we going to repeat all this in every article that's about a particular DDS? Maybe we should say it in one article and then just link to it here.

Reply:
I like that. I can just link to the `map` article's overview on this if we think that's reasonable.